index/scorch/optimize_knn.go

```diff
@@ -89,9 +92,24 @@ func (o *OptimizeVR) Finish() error {
 					vectorIndexSize := vecIndex.Size()
 					origSeg.cachedMeta.updateMeta(field, vectorIndexSize)
 					for _, vr := range vrs {
+						eligibleVectorInternalIDs := vr.EligibleDocIDs()
+						// Only the eligible documents belonging to this segment
+						// will get filtered out.
+						// There is no way to determine which doc belongs to which segment
+						eligibleVectorInternalIDs.And(snapshotGlobalDocNums[index])
+
+						eligibleLocalDocNums := roaring.NewBitmap()
```

Comment:
Do you have any obvious reasons or see gains in making this a bitmap and then doing all the allocations down the line to convert it back into an acceptable format?